kitsune/search/v2/documents.py

```diff
@@ -90,19 +89,82 @@ def get_model(cls):
         return wiki_models.Document
 
 
-class QuestionDocument(DSLDocument):
+class UserInnerDoc(InnerDoc):
+    id = field.Keyword()
+    username = field.Keyword()
+
+    @classmethod
+    def prepare(cls, instance):
+        if not instance:
+            return None
+        return cls(id=instance.id, username=instance.username)
+
+
+class ProductInnerDoc(InnerDoc):
+    id = field.Keyword()
+    title = field.Keyword()
+    slug = field.Keyword()
+
+    @classmethod
+    def prepare(cls, instance):
+        if not instance:
+            return None
+        return cls(id=instance.id, title=instance.title, slug=instance.slug)
+
+
+class TopicInnerDoc(InnerDoc):
+    id = field.Keyword()
+    title = field.Keyword()
+    slug = field.Keyword()
+
+    @classmethod
+    def prepare(cls, instance):
+        if not instance:
+            return None
+        return cls(id=instance.id, title=instance.title, slug=instance.slug)
+
+
+class TagInnerDoc(InnerDoc):
```

Comment:
All the comments above apply here too

Reply:
Let's squash `ProductInnerDoc, TagInnerDoc, TopicInnderDoc` into a single document. Those are exactly the same for at least now. No need to redefine them 3 times.